### PR TITLE
Chat window - first draft

### PR DESCRIPTION
**Pull request by sidravi1**
Reviewer: @ziakhan04 

Estimate: 1 hour

----

# Framework for a chat window

## Ticket

Fixes: [AAQ-190](https://idinsight.atlassian.net/browse/AAQ-190?atlOrigin=eyJpIjoiNWQyMmMwNzQ3ZTExNDM4MWJmZmMyZWZmYWNmODQyOTgiLCJwIjoiaiJ9)

## Description, Motivation and Context

We want to allow admins to be able to test out reponses for messages

### Changes

Added a new page.
Updated nav bar.
Some minor cosmetic changes.

## Next steps

* Make prettier
* Allow toggling - raw response, top n answers, llm response
* Fix hacky solution for getting backend token

## How Has This Been Tested?

Clicked around (though we do need to get better at this)

## Checklist:

This checklist is a useful reminder of small things that can easily be forgotten.
Put an `x` in all the items that apply and remove any items that are not relevant to this PR.

- [x] My code follows the style guidelines of this project
- [x] I have reviewed my own code to ensure good quality
- [x] I have tested the functionality of my code to ensure it works as intended
- [x] I have resolved merge conflicts
- [ ] I have updated the automated tests (if applicable)
- [x] I have written [good commit messages][1]
- [ ] I have updated the README file (if applicable)
- [ ] I have updated affected documentation (if applicable)

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: admin_app/src/app/layout.tsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import "./globals.css";
 import type { Metadata } from "next";
 import { Inter } from "next/font/google";
+import { NavBar } from "../components/NavBar";
```
Comment (sidravi1) [inline admin_app/src/app/layout.tsx:4]: Now it's added to all pages automatically

File: admin_app/src/app/page.tsx [hunk 2]
```diff
@@ -3,13 +3,10 @@
...
 import { ContentCard, Content } from "../components/ContentCard";
 import { ConfirmDelete, EditModal } from "../components/ContentModals";
-import { NavBar } from "../components/NavBar";
 import { SearchBar } from "../components/SearchBar";
 import { jwtDecode } from "jwt-decode";
 import IsFullAccess from "../components/Auth";
-
-const backendUrl: string =
-  process.env.NEXT_PUBLIC_BACKEND_URL || "http://localhost:8000";
+import { backendUrl } from "../components/Config";
```
Comment (sidravi1) [inline admin_app/src/app/page.tsx:9]: new file for all the config so we are not repeating this line everywhere

File: admin_app/src/components/ChatComponents.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+    const queryMessage: Message = {
+      type: "human",
+      content: formJson.query_text.toString(),
+    };
+
+    setMessages([...messages, queryMessage]);
+
+    // pop-up to get token if no token present
+    let token: string | null = localStorage.getItem("apiToken");
```
Comment (sidravi1) [inline admin_app/src/components/ChatComponents.tsx:36]: This is a very hacky solution. Something I'll fix in the next PR

**Review by ziakhan04 — CHANGES_REQUESTED** [review]
Looks good and works!

One question - Should both fullaccess and readonly get access to chat (happening right now)?

Added minor comments.

Suggested changes -
- The background colour doesn't match across content and playground pages
- Navbar does not correctly show which page I am on. Always defaults to highlighting "Home".

Other stuff that I made tickets out of (not necessarily related to this ticket) -
- Hydration error occurs when I login and log-out. Don't know what it is. (screenshot)
- When the screen is narrow, navbar becomes three line which are non-responsive. (screenshot)
- Pressing "Enter" in the chat to send message instead of manually clicking "Send".
- AI response has area for improvement visually. (No screenshot, don't want the developer to get a heart-attack)

File: admin_app/src/components/ChatComponents.tsx (hunk 3 above)
Comment (sidravi1) [reply admin_app/src/components/ChatComponents.tsx:36]: Yeah! Problems are: 
* there is no way to edit your token once you have entered it
* so if you enter is wrong, you may have to restart your browser 🙈 

There is the cosmetic aspect of it but also the functional. I'll fix it in the next PR.

File: admin_app/src/components/NavBar.tsx [hunk 4]
```diff
@@ -48,15 +50,15 @@ export const NavBar = () => {
...
+                href="/"
                 className="block py-2 pl-3 pr-4 text-gray-900 rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-blue-700 md:p-0 dark:text-white md:dark:hover:text-blue-500 dark:hover:bg-gray-700 dark:hover:text-white md:dark:hover:bg-transparent"
               >
                 Manage Content
               </a>
             </li>
             <li>
               <a
-                href="#"
+                href="/chat"
```
[the remark this answers is not in the document]
Comment (sidravi1) [reply admin_app/src/components/NavBar.tsx:61]: Yeah there is another ticket called "make nav bar functional" or something similar. I threw the navbar together as a placeholder and not fully functional.

File: admin_app/src/app/chat/page.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  //     { type: "ai", content: "This is the tenth row of content" },
+  //     { type: "human", content: "This is the eleventh row of content" },
+  //     { type: "ai", content: "This is the twelfth row of content" },
+  //     { type: "human", content: "This is the thirteenth row of content" },
+  //   ]);
+  // }, []);
+
+  return (
+    <>
+      <main className="flex flex-grow justify-center h-full dark:bg-gray-900 py-4 border-neutral-800 rounded-md">
```
[the remark this answers is not in the document]
Comment (sidravi1) [reply admin_app/src/app/chat/page.tsx:41]: Thanks for picking this up. Fixed so they both inherit background from body in both dark and light mode.

File: admin_app/src/components/ChatComponents.tsx [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+  };
+
+  return (
+    <form
+      className="flex absolute bottom-0 left-0 w-full px-2 space-x-2"
+      method="post"
+      onSubmit={handleSubmit}
+    >
+      <TextAreaAutosize
+        className="bg-gray-200 dark:bg-gray-800 p-2 flex-1 rounded-md border border-red-200"
```
[the remark this answers is not in the document]
Comment (sidravi1) [reply admin_app/src/components/ChatComponents.tsx:82]: No you're right. Made it something more neutral

**Pull request merged**